From 6a63ba971edbbfae280ea0cc4502e7ba4354125b Mon Sep 17 00:00:00 2001
From: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
Date: Sat, 11 Jul 2020 16:53:43 +0300
Subject: [PATCH] Revert "serial: amba-pl011: Make sure we initialize the port.lock spinlock"
Git-commit: 6a63ba971edbbfae280ea0cc4502e7ba4354125b
Patch-mainline: v5.9-rc1
References: git-fixes

This reverts commit 8508f4cba308f785b2fd4b8c38849c117b407297.

There has been a quick fix against uninitialised lock revealed by
the commit f743061a85f5 ("serial: core: Initialise spin lock before use
in uart_configure_port()"). Since we have now better fix in serial core,
this may be safely reverted.

Fixes: 8508f4cba308 ("serial: amba-pl011: Make sure we initialize the port.lock spinlock")
Depends-on: f743061a85f5 ("serial: core: Initialise spin lock before use in uart_configure_port()")
Signed-off-by: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
Link: https://lore.kernel.org/r/20200711135346.71171-2-andriy.shevchenko@linux.intel.com
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 drivers/tty/serial/amba-pl011.c |    1 -
 1 file changed, 1 deletion(-)

--- a/drivers/tty/serial/amba-pl011.c
+++ b/drivers/tty/serial/amba-pl011.c
@@ -2604,7 +2604,6 @@ static int pl011_setup_port(struct devic
 	uap->port.fifosize = uap->fifosize;
 	uap->port.flags = UPF_BOOT_AUTOCONF;
 	uap->port.line = index;
-	spin_lock_init(&uap->port.lock);
 
 	amba_ports[index] = uap;
 
